### ISSUE TITLE
[Feature] Ability to use templates in go-select

### ISSUE THREAD

**Issue by StevenUlmer** [mobi/goponents#375]
### What is the feature that you would like to propose?
There are times where we want to display an option or a selected option with more than just one attribute in the select.

For example, we have a user that has a first and last name and we want to display the option to the user as `First Last`. 

Showing just the first or last name without the other doesn't give the user enough context to make a decision on which option to pick and it can be a pain to go through an array of items and map out an attribute for each of those items that is what we want to display.

### Describe the solution you'd like
Use an ng-template to define what we want the user to see. Something like the following. This would show the first and last name for both the options and the selected option. We should be able to use the ng-select `ng-option-tmp` and `ng-label-tmp` templates in order to achieve this
```
<go-select [control]="itemControl">
  <ng-template go-select-option let-item="item">
     {{ item.firstName }} - {{ item.lastName }}
  </ng-template>
</go-select>
```

**Comment by grahamhency** [comment]
@StevenUlmer I believe @mwagz already has created something like this with the `go-lookup` in the main mobi project.

**Comment by grahamhency** [comment]
Closed with #425
